visual-recognition/src/main/java/com/ibm/watson/developer_cloud/visual_recognition/v3/VisualRecognition.java

```diff
@@ -97,6 +110,9 @@ protected void setAuthentication(okhttp3.Request.Builder builder) {
     if (getApiKey() == null) {
       throw new IllegalArgumentException("api_key needs to be specified. Use setApiKey()");
     }
+    if (endPointChanged && !getEndPoint().equals(URL)) {
+      super.setAuthentication(builder);
```

Comment:
Would it be wrong to set this authentication header even if the request is targeting a public instance of VR?

If not, then why not just supply it always and keep the code simple.

But if it is a problem, then I worry that the endpoint might be changed because it is for a service instance in a different region.  In that case, we don't want to provide this authorization header.

Reply:
Yeah we only want to set the authentication header if it's a dedicated instance. Otherwise, the service wants the API key as a query parameter.

You're right though that there might be cases where the endpoint is changed but isn't pointing to a dedicated instance.. Is there any way to check whether or not the user is trying to use one?

Reply:
I think my preference would be to change the WatsonService so that it stores the username/password separately from the APIKey.  If we do this, then we can prefer the username/password whenever it it specified, and fall back to APIKey (for VR) otherwise.

Reply:
Agreed. I'll look at making that change now.